### PR TITLE
feat(consensus): CON-675 Increase notary delay caused by certified-finalized gap

### PR DESCRIPTION
**Pull request by eichhorl**
Recently, we are observing multiple subnets maintaining a (relatively) constant gap of 3-5 rounds between their latest finalized and certified heights, which likely leads to higher end-to-end ingress latencies on these subnets.

With this PR, we attempt to reduce the size of this gap, by increasing the notary delay more aggressively once the "acceptable gap" between finalization and certification is exceeded. This is done by including the heights _within_ the acceptable gap when calculating the additive linear delay. 

Additionally, instead of using the unit delay (1s on app subnets, 3s on large/system subnets) to calculate the delay, we switch to a new "backlog delay" which is set to a constant 2 seconds. The unit delay isn't related to the speed of execution, so having two values for it is confusing. 

Furthermore, we 
- reduce the acceptable gap from 3 to 2,
- fix a test to calculate the expected delay correctly.

Running the `consensus_performance_test` with an added execution delay of 2s gives the following results:
- Certified-finalized gap: 4\~5 (old implementation) -> 2\~3 (new implementation)
- Average E2E ingress latency: 10.2s (old implementation) -> 8.8s (new implementation)

chart (1)

### CODE REVIEW

**Comment by kpop-dfinity** [comment]
Did you run `consensus_performance_test` by any chance?

**Comment by eichhorl** [comment]
> Did you run `consensus_performance_test` by any chance?

Yes, multiple times 😄. By default there are no differences in the test results when comparing both implementations (likely because the certified-finalized gap is very low at all times). But after adding a 2s delay to execution,I could see the gap and E2E ingress latency go down with the new implementation.

**Comment by kpop-dfinity** [comment]
> > Did you run `consensus_performance_test` by any chance?
> 
> Yes, multiple times 😄. By default there are no differences in the test results when comparing both implementations (likely because the certified-finalized gap is very low at all times). But after adding a 2s delay to execution,I could see the gap and E2E ingress latency go down with the new implementation.

I just saw your message on slack :D. Thanks!
Maybe it would be a good idea to include some of these findings in the PR description?

File: rs/consensus/src/consensus/notary.rs [hunk 1]
```diff
@@ -49,7 +49,11 @@ use std::{sync::Arc, time::Duration};
 /// The acceptable gap between the finalized height and the certified height. If
 /// the actual gap is greater than this, consensus starts slowing down the block
 /// rate.
-const ACCEPTABLE_FINALIZATION_CERTIFICATION_GAP: u64 = 3;
+const ACCEPTABLE_FINALIZATION_CERTIFICATION_GAP: u64 = 2;
+
+/// The amount of time consensus should delay notarization of the next block by,
+/// for each height that the latest finalized block is ahead of the latest certified state.
+const BACKLOG_DELAY_MILLIS: u64 = 2000;
```
Comment (kpop-dfinity) [inline rs/consensus/src/consensus/notary.rs:56]: nit:
```suggestion
const BACKLOG_DELAY_MILLIS: u64 = 2_000;
```

Comment (kpop-dfinity) [inline rs/consensus/src/consensus/notary.rs:56]: Should we explain here how this value was chosen?

**Review by kpop-dfinity — APPROVED** [review]

File: rs/consensus/src/consensus/notary.rs (hunk 1 above)
Comment (eichhorl) [reply rs/consensus/src/consensus/notary.rs]: I was trying to keep the slope somewhat similar to what we had before (1s & 3s) to be cautious. Empirically, 2s also worked better then 1s, but this is something we can adjust further if needed. I updated the comment

**Pull request merged**